address-space-controller/src/main/java/io/enmasse/controller/auth/CertBundleCertProvider.java

```diff
@@ -41,8 +41,8 @@ public void provideCert(AddressSpace addressSpace, EndpointInfo endpointInfo) {
             return;
         }
 
-        data.put("tls.key", tlsKey);
-        data.put("tls.crt", tlsCert);
+        data.put("tls.key", tlsKey.replace("\n", ""));
```

Comment:
I'm not sure I understand the intent by stripping the newlines?  I think the data of cert is PEM formatted, so the newlines (which are allowed to be an line ending convention) are required by the format.

Reply:
This is base64 encoded. If the input is taken as a newline formatted base64 (which is common if you use the base64 command line tools etc), comparing desired vs current state will always trigger a replace because kubernetes strips the newlines when creating the secret.

Reply:
The input is the base64 encoded string of the PEM, i.e. almost encoded twice, which made me think maybe we should change the expected input to be the PEM itelf instead requiring it to be base64-encoded?

Reply:
@k-wall I think the source of confusion here is that the input format was a base64 encoded string of the original PEM. But that is not really required, so I pushed a fix to allow the input to be PEM-formatted as well.